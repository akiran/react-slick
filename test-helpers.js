export function repeatClicks(node, count) {
  for (let i = 0; i < count; i++) {
    node.simulate("click");
  }
}

export function delay(duration) {
  return new Promise(function(resolve) {
    setTimeout(resolve, duration);
  });
}
