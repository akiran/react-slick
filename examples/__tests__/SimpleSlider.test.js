import React from 'react';
import { mount } from 'enzyme';
import SimpleSlider from '../SimpleSlider';
import { repeatClicks } from '../../test-helpers';
import { html as beautify_html } from 'js-beautify'

describe('Simple Slider', function () {
  it('should have 13 slides (1(preclone) + 6(actual) + 6(postclone))', function () {
    const wrapper = mount(<SimpleSlider />);
    expect(wrapper.find('.slick-slide').length).toEqual(13);
  });
  it('should have 7 clone slides', function () {
    const wrapper = mount(<SimpleSlider />);
    expect(wrapper.find('.slick-cloned').length).toEqual(7);
  });
  it('should have 1 active slide', function () {
    const wrapper = mount(<SimpleSlider />);
    expect(wrapper.find('.slick-slide.slick-active').length).toEqual(1);
  });
  it('should have 6 dots', function () {
    const wrapper = mount(<SimpleSlider />);
    expect(wrapper.find('.slick-dots').children().length).toEqual(6);
  });
  it('should have 1 active dot', function () {
    const wrapper = mount(<SimpleSlider />);
    expect(wrapper.find('.slick-dots .slick-active').length).toEqual(1);
  });
  it('should have a prev arrow', function () {
    const wrapper = mount(<SimpleSlider />);
    expect(wrapper.find('.slick-prev').length).toEqual(1);
  });
  it('should have a next arrow', function () {
    const wrapper = mount(<SimpleSlider />);
    expect(wrapper.find('.slick-next').length).toEqual(1);
  });

  it('should got to second slide when next button is clicked', function () {
    const wrapper = mount(<SimpleSlider />);
    wrapper.find('.slick-next').simulate('click')
    expect(wrapper.find('.slick-slide.slick-active').first().text()).toEqual('2');
    expect(wrapper.find('.slick-dots .slick-active').length).toEqual(1);
    expect(wrapper.find('.slick-dots').childAt(1).hasClass('slick-active')).toEqual(true)
  });
  it('should goto last slide when prev button is clicked', function () {
    const wrapper = mount(<SimpleSlider />);
    wrapper.find('.slick-prev').simulate('click')
    expect(wrapper.find('.slick-slide.slick-active').first().text()).toEqual('6');
    expect(wrapper.find('.slick-dots .slick-active').length).toEqual(1);
    expect(wrapper.find('.slick-dots').childAt(5).hasClass('slick-active')).toEqual(true)
  })
  it('should goto 4th slide when 4th dot is clicked', function () {
    const wrapper = mount(<SimpleSlider />);
    wrapper.find('.slick-dots button').at(3).simulate('click')
    expect(wrapper.find('.slick-slide.slick-active').first().text()).toEqual('4');
    expect(wrapper.find('.slick-dots .slick-active').length).toEqual(1);
    expect(wrapper.find('.slick-dots').childAt(3).hasClass('slick-active')).toEqual(true)
  })
  // it('should come back to 1st slide after 6 clicks on next button', function () {
  //   // waitForAnimate option is causing problem for this test
  //   const wrapper = mount(<SimpleSlider />);
  //   wrapper.find('.slick-next').first().simulate('click').simulate('click')
  //   // wrapper.find('.slick-prev').first().simulate('click')
  //   // wrapper.find('.slick-next').first().simulate('click')
  //   // console.log(nextButton)
  //   // nextButton.simulate('click').simulate('click')
  //   // nextButton.simulate('click')
  //   // repeatClicks(wrapper.find('.slick-next'), 1)
  //   expect(wrapper.find('.slick-slide.slick-active').first().text()).toEqual('1');
  //   expect(wrapper.find('.slick-dots .slick-active').length).toEqual(1);
  //   expect(wrapper.find('.slick-dots').childAt(0).hasClass('slick-active')).toEqual(true)
  // })
});

describe("Simple Slider Snapshots", function () {
  it("slider initial state", function () {
    const wrapper = mount(<SimpleSlider />);
    expect(beautify_html(wrapper.html())).toMatchSnapshot()
  });
  it("click on next button", function () {
    const wrapper = mount(<SimpleSlider />);
    wrapper.find('.slick-next').simulate('click')
    expect(beautify_html(wrapper.html())).toMatchSnapshot()
  });
  it("click on prev button", function () {
    const wrapper = mount(<SimpleSlider />);
    wrapper.find('.slick-prev').simulate('click')
    expect(beautify_html(wrapper.html())).toMatchSnapshot()
  });
  it("click on 3rd dot", function () {
    const wrapper = mount(<SimpleSlider />);
    wrapper.find('.slick-dots button').at(2).simulate('click')
    expect(beautify_html(wrapper.html())).toMatchSnapshot()
  })
});

