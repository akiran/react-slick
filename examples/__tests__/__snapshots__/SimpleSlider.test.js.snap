// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Simple Slider Snapshots click on 3rd dot 1`] = `
"<div>
    <h2> Single Item</h2>
    <div class=\\"slick-slider slick-initialized\\" dir=\\"ltr\\"><button type=\\"button\\" data-role=\\"none\\" class=\\"slick-arrow slick-prev\\" style=\\"display: block;\\"> Previous</button>
        <div class=\\"slick-list\\">
            <div aria-labelledby=\\"slickSliderLabel slickSliderInst\\" role=\\"list\\" class=\\"slick-track\\" style=\\"opacity: 1;\\">
                <div data-index=\\"-1\\" class=\\"slick-slide slick-cloned\\" aria-hidden=\\"true\\" style=\\"width: 0px;\\"></div>
                <div data-index=\\"0\\" role=\\"listitem\\" class=\\"slick-slide\\" style=\\"outline: none; width: 0px;\\">
                    <div>
                        <div style=\\"width: 100%; display: inline-block;\\">
                            <h3>1</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"1\\" role=\\"listitem\\" class=\\"slick-slide\\" style=\\"outline: none; width: 0px;\\"></div>
                <div data-index=\\"2\\" role=\\"listitem\\" class=\\"slick-slide slick-active slick-current\\" style=\\"outline: none; width: 0px;\\">
                    <div>
                        <div style=\\"width: 100%; display: inline-block;\\">
                            <h3>3</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"3\\" role=\\"listitem\\" class=\\"slick-slide\\" style=\\"outline: none; width: 0px;\\"></div>
                <div data-index=\\"4\\" role=\\"listitem\\" class=\\"slick-slide\\" style=\\"outline: none; width: 0px;\\"></div>
                <div data-index=\\"5\\" role=\\"listitem\\" class=\\"slick-slide\\" style=\\"outline: none; width: 0px;\\"></div>
                <div data-index=\\"6\\" class=\\"slick-slide slick-cloned\\" aria-hidden=\\"true\\" style=\\"width: 0px;\\">
                    <div>
                        <div style=\\"width: 100%; display: inline-block;\\">
                            <h3>1</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"7\\" class=\\"slick-slide slick-cloned\\" aria-hidden=\\"true\\" style=\\"width: 0px;\\"></div>
                <div data-index=\\"8\\" class=\\"slick-slide slick-cloned\\" aria-hidden=\\"true\\" style=\\"width: 0px;\\">
                    <div>
                        <div style=\\"width: 100%; display: inline-block;\\">
                            <h3>3</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"9\\" class=\\"slick-slide slick-cloned\\" aria-hidden=\\"true\\" style=\\"width: 0px;\\"></div>
                <div data-index=\\"10\\" class=\\"slick-slide slick-cloned\\" aria-hidden=\\"true\\" style=\\"width: 0px;\\"></div>
                <div data-index=\\"11\\" class=\\"slick-slide slick-cloned\\" aria-hidden=\\"true\\" style=\\"width: 0px;\\"></div>
            </div>
        </div><button type=\\"button\\" data-role=\\"none\\" class=\\"slick-arrow slick-next\\" style=\\"display: block;\\"> Next</button>
        <ul style=\\"display: block;\\" class=\\"slick-dots\\">
            <li class=\\"\\"><button>1</button></li>
            <li class=\\"\\"><button>2</button></li>
            <li class=\\"slick-active\\"><button>3</button></li>
            <li class=\\"\\"><button>4</button></li>
            <li class=\\"\\"><button>5</button></li>
            <li class=\\"\\"><button>6</button></li>
        </ul>
    </div>
</div>"
`;

exports[`Simple Slider Snapshots click on next button 1`] = `
"<div>
    <h2> Single Item</h2>
    <div class=\\"slick-slider slick-initialized\\" dir=\\"ltr\\"><button type=\\"button\\" data-role=\\"none\\" class=\\"slick-arrow slick-prev\\" style=\\"display: block;\\"> Previous</button>
        <div class=\\"slick-list\\">
            <div aria-labelledby=\\"slickSliderLabel slickSliderInst\\" role=\\"list\\" class=\\"slick-track\\" style=\\"opacity: 1;\\">
                <div data-index=\\"-1\\" class=\\"slick-slide slick-cloned\\" aria-hidden=\\"true\\" style=\\"width: 0px;\\"></div>
                <div data-index=\\"0\\" role=\\"listitem\\" class=\\"slick-slide\\" style=\\"outline: none; width: 0px;\\">
                    <div>
                        <div style=\\"width: 100%; display: inline-block;\\">
                            <h3>1</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"1\\" role=\\"listitem\\" class=\\"slick-slide slick-active slick-current\\" style=\\"outline: none; width: 0px;\\">
                    <div>
                        <div style=\\"width: 100%; display: inline-block;\\">
                            <h3>2</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"2\\" role=\\"listitem\\" class=\\"slick-slide\\" style=\\"outline: none; width: 0px;\\"></div>
                <div data-index=\\"3\\" role=\\"listitem\\" class=\\"slick-slide\\" style=\\"outline: none; width: 0px;\\"></div>
                <div data-index=\\"4\\" role=\\"listitem\\" class=\\"slick-slide\\" style=\\"outline: none; width: 0px;\\"></div>
                <div data-index=\\"5\\" role=\\"listitem\\" class=\\"slick-slide\\" style=\\"outline: none; width: 0px;\\"></div>
                <div data-index=\\"6\\" class=\\"slick-slide slick-cloned\\" aria-hidden=\\"true\\" style=\\"width: 0px;\\">
                    <div>
                        <div style=\\"width: 100%; display: inline-block;\\">
                            <h3>1</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"7\\" class=\\"slick-slide slick-cloned\\" aria-hidden=\\"true\\" style=\\"width: 0px;\\">
                    <div>
                        <div style=\\"width: 100%; display: inline-block;\\">
                            <h3>2</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"8\\" class=\\"slick-slide slick-cloned\\" aria-hidden=\\"true\\" style=\\"width: 0px;\\"></div>
                <div data-index=\\"9\\" class=\\"slick-slide slick-cloned\\" aria-hidden=\\"true\\" style=\\"width: 0px;\\"></div>
                <div data-index=\\"10\\" class=\\"slick-slide slick-cloned\\" aria-hidden=\\"true\\" style=\\"width: 0px;\\"></div>
                <div data-index=\\"11\\" class=\\"slick-slide slick-cloned\\" aria-hidden=\\"true\\" style=\\"width: 0px;\\"></div>
            </div>
        </div><button type=\\"button\\" data-role=\\"none\\" class=\\"slick-arrow slick-next\\" style=\\"display: block;\\"> Next</button>
        <ul style=\\"display: block;\\" class=\\"slick-dots\\">
            <li class=\\"\\"><button>1</button></li>
            <li class=\\"slick-active\\"><button>2</button></li>
            <li class=\\"\\"><button>3</button></li>
            <li class=\\"\\"><button>4</button></li>
            <li class=\\"\\"><button>5</button></li>
            <li class=\\"\\"><button>6</button></li>
        </ul>
    </div>
</div>"
`;

exports[`Simple Slider Snapshots click on prev button 1`] = `
"<div>
    <h2> Single Item</h2>
    <div class=\\"slick-slider slick-initialized\\" dir=\\"ltr\\"><button type=\\"button\\" data-role=\\"none\\" class=\\"slick-arrow slick-prev\\" style=\\"display: block;\\"> Previous</button>
        <div class=\\"slick-list\\">
            <div aria-labelledby=\\"slickSliderLabel slickSliderInst\\" role=\\"list\\" class=\\"slick-track\\" style=\\"opacity: 1;\\">
                <div data-index=\\"-1\\" class=\\"slick-slide slick-cloned\\" aria-hidden=\\"true\\" style=\\"width: 0px;\\">
                    <div>
                        <div style=\\"width: 100%; display: inline-block;\\">
                            <h3>6</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"0\\" role=\\"listitem\\" class=\\"slick-slide\\" style=\\"outline: none; width: 0px;\\">
                    <div>
                        <div style=\\"width: 100%; display: inline-block;\\">
                            <h3>1</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"1\\" role=\\"listitem\\" class=\\"slick-slide\\" style=\\"outline: none; width: 0px;\\"></div>
                <div data-index=\\"2\\" role=\\"listitem\\" class=\\"slick-slide\\" style=\\"outline: none; width: 0px;\\"></div>
                <div data-index=\\"3\\" role=\\"listitem\\" class=\\"slick-slide\\" style=\\"outline: none; width: 0px;\\"></div>
                <div data-index=\\"4\\" role=\\"listitem\\" class=\\"slick-slide\\" style=\\"outline: none; width: 0px;\\"></div>
                <div data-index=\\"5\\" role=\\"listitem\\" class=\\"slick-slide slick-active slick-current\\" style=\\"outline: none; width: 0px;\\">
                    <div>
                        <div style=\\"width: 100%; display: inline-block;\\">
                            <h3>6</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"6\\" class=\\"slick-slide slick-cloned\\" aria-hidden=\\"true\\" style=\\"width: 0px;\\">
                    <div>
                        <div style=\\"width: 100%; display: inline-block;\\">
                            <h3>1</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"7\\" class=\\"slick-slide slick-cloned\\" aria-hidden=\\"true\\" style=\\"width: 0px;\\"></div>
                <div data-index=\\"8\\" class=\\"slick-slide slick-cloned\\" aria-hidden=\\"true\\" style=\\"width: 0px;\\"></div>
                <div data-index=\\"9\\" class=\\"slick-slide slick-cloned\\" aria-hidden=\\"true\\" style=\\"width: 0px;\\"></div>
                <div data-index=\\"10\\" class=\\"slick-slide slick-cloned\\" aria-hidden=\\"true\\" style=\\"width: 0px;\\"></div>
                <div data-index=\\"11\\" class=\\"slick-slide slick-cloned\\" aria-hidden=\\"true\\" style=\\"width: 0px;\\">
                    <div>
                        <div style=\\"width: 100%; display: inline-block;\\">
                            <h3>6</h3>
                        </div>
                    </div>
                </div>
            </div>
        </div><button type=\\"button\\" data-role=\\"none\\" class=\\"slick-arrow slick-next\\" style=\\"display: block;\\"> Next</button>
        <ul style=\\"display: block;\\" class=\\"slick-dots\\">
            <li class=\\"\\"><button>1</button></li>
            <li class=\\"\\"><button>2</button></li>
            <li class=\\"\\"><button>3</button></li>
            <li class=\\"\\"><button>4</button></li>
            <li class=\\"\\"><button>5</button></li>
            <li class=\\"slick-active\\"><button>6</button></li>
        </ul>
    </div>
</div>"
`;

exports[`Simple Slider Snapshots slider initial state 1`] = `
"<div>
    <h2> Single Item</h2>
    <div class=\\"slick-slider slick-initialized\\" dir=\\"ltr\\"><button type=\\"button\\" data-role=\\"none\\" class=\\"slick-arrow slick-prev\\" style=\\"display: block;\\"> Previous</button>
        <div class=\\"slick-list\\">
            <div aria-labelledby=\\"slickSliderLabel slickSliderInst\\" role=\\"list\\" class=\\"slick-track\\" style=\\"opacity: 1;\\">
                <div data-index=\\"-1\\" class=\\"slick-slide slick-cloned\\" aria-hidden=\\"true\\" style=\\"width: 0px;\\"></div>
                <div data-index=\\"0\\" role=\\"listitem\\" class=\\"slick-slide slick-active slick-current\\" style=\\"outline: none; width: 0px;\\">
                    <div>
                        <div style=\\"width: 100%; display: inline-block;\\">
                            <h3>1</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"1\\" role=\\"listitem\\" class=\\"slick-slide\\" style=\\"outline: none; width: 0px;\\"></div>
                <div data-index=\\"2\\" role=\\"listitem\\" class=\\"slick-slide\\" style=\\"outline: none; width: 0px;\\"></div>
                <div data-index=\\"3\\" role=\\"listitem\\" class=\\"slick-slide\\" style=\\"outline: none; width: 0px;\\"></div>
                <div data-index=\\"4\\" role=\\"listitem\\" class=\\"slick-slide\\" style=\\"outline: none; width: 0px;\\"></div>
                <div data-index=\\"5\\" role=\\"listitem\\" class=\\"slick-slide\\" style=\\"outline: none; width: 0px;\\"></div>
                <div data-index=\\"6\\" class=\\"slick-slide slick-cloned\\" aria-hidden=\\"true\\" style=\\"width: 0px;\\">
                    <div>
                        <div style=\\"width: 100%; display: inline-block;\\">
                            <h3>1</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"7\\" class=\\"slick-slide slick-cloned\\" aria-hidden=\\"true\\" style=\\"width: 0px;\\"></div>
                <div data-index=\\"8\\" class=\\"slick-slide slick-cloned\\" aria-hidden=\\"true\\" style=\\"width: 0px;\\"></div>
                <div data-index=\\"9\\" class=\\"slick-slide slick-cloned\\" aria-hidden=\\"true\\" style=\\"width: 0px;\\"></div>
                <div data-index=\\"10\\" class=\\"slick-slide slick-cloned\\" aria-hidden=\\"true\\" style=\\"width: 0px;\\"></div>
                <div data-index=\\"11\\" class=\\"slick-slide slick-cloned\\" aria-hidden=\\"true\\" style=\\"width: 0px;\\"></div>
            </div>
        </div><button type=\\"button\\" data-role=\\"none\\" class=\\"slick-arrow slick-next\\" style=\\"display: block;\\"> Next</button>
        <ul style=\\"display: block;\\" class=\\"slick-dots\\">
            <li class=\\"slick-active\\"><button>1</button></li>
            <li class=\\"\\"><button>2</button></li>
            <li class=\\"\\"><button>3</button></li>
            <li class=\\"\\"><button>4</button></li>
            <li class=\\"\\"><button>5</button></li>
            <li class=\\"\\"><button>6</button></li>
        </ul>
    </div>
</div>"
`;
